### ISSUE TITLE
Don't report multiple-statements for def + ellipsis

### ISSUE THREAD

**Issue by mthuurne** [PyCQA/pylint#3224]
Code using type annotations often has no need for a function body and puts ellipsis (`...`) there to indicate that.

For example when using `@overload`:
(this was taken straight from PEP 484)

```py
from typing import overload

@overload
def concat2(x: str, y: str) -> str: ...
@overload
def concat2(x: bytes, y: bytes) -> bytes: ...
```

Or when defining a protocol:
(when using a Python version before 3.8, you'll have to install `typing_extensions` and import `Protocol` from there)

```py
from typing import Any, Protocol

class ComparableProto(Protocol):
    def __eq__(self, other: Any) -> bool: ...
    def __ne__(self, other: Any) -> bool: ...
    def __lt__(self, other: Any) -> bool: ...
    def __le__(self, other: Any) -> bool: ...
    def __gt__(self, other: Any) -> bool: ...
    def __ge__(self, other: Any) -> bool: ...
```
However, when using this syntax, PyLint 2.4.3 will report `More than one statement on a single line (multiple-statements)`.

While it is technically correct that the ellipsis is a statement, it is used to indicate the absence of meaningful statements. So in my opinion it would be better to have a special case for not reporting the situation in which there are two statements, the first of which is `def` and the second of which is `Ellipsis`.

**Comment by PCManticore** [comment]
Yeah, I think we can exempt these situations.

**Comment by mthuurne** [comment]
Thanks!

**Comment by mthuurne** [comment]
On my PC, the fix works for Protocols, but for some reason not for functions/methods. Even the following fragment from the unit test issues a `multiple-statements` message:
```py
from typing import overload
@overload
def concat2(arg1: str) -> str: ...
```
These are the versions I'm using:
```
pylint 2.4.4
astroid 2.3.3
Python 3.7.2 (default, Feb 26 2019, 13:02:33) 
[GCC 7.3.1 20180323 [gcc-7-branch revision 258812]]
```
